spec/presenters/coronavirus/sub_section_json_presenter_spec.rb

```diff
@@ -158,6 +154,142 @@
     end
   end
 
+  describe "#sub_sections" do
+    let(:title) { Faker::Lorem.sentence }
+    let(:label) { Faker::Lorem.sentence }
+    let(:path)  { "/#{File.join(Faker::Lorem.words)}" }
+    let(:sub_section) do
+      create :coronavirus_sub_section, content: "###{title}\n[#{label}](#{path})"
+    end
+
+    it "includes an action link if one exists" do
+      sub_section.action_link_url = "/bananas"
+      sub_section.action_link_content = "Bananas"
+      sub_section.action_link_summary = "Bananas"
+
+      expected = [
+        {
+          list: [
+            {
+              url: "/bananas",
+              label: "Bananas",
+              description: "Bananas",
+              featured_link: true,
+            },
+          ],
+          title: nil,
+        },
+        {
+          list: [
+            {
+              url: path,
+              label: label,
+            },
+          ],
+          title: title,
+        },
+      ]
+
+      expect(subject.sub_sections).to eq(expected)
+    end
+
+    it "does not include an action link if there isn't one" do
+      expected = [
+        {
+          list: [
+            {
+              url: path,
+              label: label,
+            },
+          ],
+          title: title,
+        },
+      ]
+
+      expect(subject.sub_sections).to eq(expected)
+    end
+
+    context "priority_taxon" do
+      let(:priority_taxon) { SecureRandom.uuid }
+      let(:content) { Faker::Lorem.sentence }
+      let(:summary) { Faker::Lorem.sentence }
+      let(:sub_section) do
+        build(:coronavirus_sub_section,
+              content: "#Title",
+              action_link_content: content,
+              action_link_summary: summary)
+      end
+      let(:presenter) { described_class.new(sub_section, priority_taxon) }
+
+      it "includes an action link with a priority_taxon if the link is relative" do
+        sub_section.action_link_url = "/bananas"
+        sub_sections_list = presenter.output[:sub_sections].first[:list]
+
+        expect(sub_sections_list).to include(
+          hash_including(
+            url: "/bananas?priority-taxon=#{priority_taxon}",
+            label: content,
+            description: summary,
+            featured_link: true,
+          ),
+        )
+      end
+
+      it "does not include an action link with a priority_taxon if the link is external" do
+        sub_section.action_link_url = "https://www.bananas.com"
+        sub_sections_list = presenter.output[:sub_sections].first[:list]
+
+        expect(sub_sections_list).to include(
+          hash_including(
+            url: "https://www.bananas.com",
+            label: content,
+            description: summary,
+            featured_link: true,
+          ),
+        )
+      end
+    end
+  end
+
+  describe "#append_priority_taxon_query_param" do
```

Comment:
I don't think these tests are needed as this code is already tested through `#sub_section` tests. The only case missing is the no taxon one which I've suggested you add in above.